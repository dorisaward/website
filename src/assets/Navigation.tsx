import { Link } from 'react-router-dom'
import styled from 'styled-components'

const StyledNav = styled.nav`
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    height: 70%;

    a:link { text-decoration: none; }
    a:visited { text-decoration: none; }
    a:hover { text-decoration: none; }
    a:active { text-decoration: none; }
`

const LinkContainer = styled.div`
    padding: ${({theme}) => theme.padding};
    margin: ${({theme}) => theme.padding};
    background-color: ${({theme}) => theme.tertiaryColour};
    border-radius: ${({theme}) => theme.padding};
    box-shadow: ${({theme}) => '1px 1px ' + theme.shadowColour};

    &:active { box-shadow: unset; }
`

export const Navigation = () => {
    return (
        <StyledNav>
            <LinkContainer><Link to="/">Home</Link></LinkContainer>
            <LinkContainer><Link to="/cv">Curriculum Vitae</Link></LinkContainer>
        </StyledNav>
    )
}
