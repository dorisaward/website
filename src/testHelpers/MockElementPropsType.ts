import { ReactNode } from 'react'

export type MockElementPropsType<P = unknown> = P & { children?: ReactNode | undefined }
