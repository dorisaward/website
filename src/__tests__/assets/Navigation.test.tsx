import { render } from '@testing-library/react'
import { Navigation } from '../../assets/Navigation.tsx'
import { MockElementPropsType } from '../../testHelpers/MockElementPropsType.ts'

jest.mock('react-router-dom', () => ({
    Link: ({ children, ...restProps }: MockElementPropsType) => <div {...restProps}>{children}</div>
}))

describe('Navigation', () => {
    it('renders correctly', () => {
        // Given
        const { baseElement, getByRole } = render(<Navigation />)

        // Then
        expect(getByRole('navigation')).toBeTruthy()
        expect(baseElement).toMatchSnapshot()
    })
})