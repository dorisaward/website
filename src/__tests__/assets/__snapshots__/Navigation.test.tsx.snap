// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<body>
  <div>
    <nav
      class="sc-blHHSb dxgyhn"
    >
      <div
        to="/"
      >
        <div
          class="sc-gtLWhw SrjuU"
        />
      </div>
      <div
        to="/cv"
      >
        <div
          class="sc-gtLWhw SrjuU"
        />
      </div>
    </nav>
  </div>
</body>
`;
